buildscripts/docker-entrypoint.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+#
+# Minio Cloud Storage, (C) 2017 Minio, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# if command starts with an option, prepend minio
+if [ "${1}" != "minio" ]; then
+    if [ -n "${1}" ]; then
+        set -- minio "$@"
+    fi
+fi
+
+## Look for swarm secrets in default location.
+file_env() {
+    local MINIO_ACCESS_KEY_FILE="/run/secrets/access_key"
+    local MINIO_SECRET_KEY_FILE="/run/secrets/secret_key"
+
+    if [ -f $MINIO_ACCESS_KEY_FILE -a -f $MINIO_SECRET_KEY_FILE ]; then 
+        if [ -f $MINIO_ACCESS_KEY_FILE ]; then
+            export MINIO_ACCESS_KEY="$(< "$MINIO_ACCESS_KEY_FILE")"
```

Comment:
Also tested with the ENVs leaking process envs run through a script are well inside the process itself so an entrypoint.sh contains it from being exposed, since these are not set container wide.

```
/data # env
HOSTNAME=1f3d3b6c405b
SHLVL=1
HOME=/root
MINIO_TEST.1.PGIKYY0C35OEJLYKI4V6XMZPW_ENV_GOPATH=/go
MINIO_TEST.1.PGIKYY0C35OEJLYKI4V6XMZPW_PORT_9000_TCP_ADDR=172.17.0.2
MINIO_TEST.1.PGIKYY0C35OEJLYKI4V6XMZPW_ENV_CGO_ENABLED=0
MINIO_TEST.1.PGIKYY0C35OEJLYKI4V6XMZPW_PORT_9000_TCP_PORT=9000
REDIS_DOWNLOAD_SHA1=6780d1abb66f33a97aad0edbe020403d0a15b67f
MINIO_TEST.1.PGIKYY0C35OEJLYKI4V6XMZPW_PORT_9000_TCP_PROTO=tcp
TERM=xterm
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
MINIO_TEST.1.PGIKYY0C35OEJLYKI4V6XMZPW_PORT_9000_TCP=tcp://172.17.0.2:9000
MINIO_TEST.1.PGIKYY0C35OEJLYKI4V6XMZPW_PORT=tcp://172.17.0.2:9000
REDIS_DOWNLOAD_URL=http://download.redis.io/releases/redis-3.2.8.tar.gz
MINIO_TEST.1.PGIKYY0C35OEJLYKI4V6XMZPW_NAME=/amazing_aryabhata/minio-test.1.pgikyy0c35oejlyki4v6xmzpw
REDIS_VERSION=3.2.8
```

```
docker run --rm --link minio-test.1.pgikyy0c35oejlyki4v6xmzpw redis:alpine
```

So even with links the secrets are not exposed. So the kernel is making sure of that.